### PR TITLE
Added `Configure Secrets` support to `databricks labs remorph configure-secrets` cli command 

### PR DESCRIPTION
**Pull request by vijaypavann-db**
## Changes

Added CLI command `databricks labs remorph configure-secrets` for creating Scope and Secrets on Databricks Workspace.

Logic:
* Prompt for Source
* Prompt for Secret Scope
* Prompt for Connection details specific to `Source`
* Store `Connection details` as `Secrets` on `Databricks Workspace`

### Linked issues
Resolves 
https://github.com/databrickslabs/remorph/issues/253

### Related issues:
https://github.com/databrickslabs/remorph/issues/209

### Functionality
- [x]  added new CLI command

### Tests
- [x]  manually tested
- [x]  added unit tests

### TODO

- [ ]  add integration tests
- [ ]  verify on staging environment (attach screenshot )

### CODE REVIEW

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import logging
+
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors.platform import ResourceDoesNotExist
+
+logger = logging.getLogger(__name__)
+
+
+class DatabricksSecretsClient:
```
Comment (nfx) [inline src/databricks/labs/remorph/helpers/db_workspace_utils.py:10]: move all of the methods from here to `ReconConfigPrompts`. You don't need to create secret scope client over an existing SDK client :)

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        if not scope_exists:
+            logger.error(
+                f"Error: Cannot find Secret Scope: `{scope_name}` in Databricks Workspace."
+                f"\nUse `remorph configure-secrets` to setup Scope and Secrets"
+            )
+            return False
+        logger.debug(f"Found Scope: `{scope_name}` in Databricks Workspace")
+        return True
+
+    def get_or_create_scope(self, scope_name: str):
```
Comment (nfx) [inline src/databricks/labs/remorph/helpers/db_workspace_utils.py:27]:
```suggestion
    def _ensure_scope_exists(self, scope_name: str):
```
better name, make private

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                logger.debug(f" Creating a new Scope: `{scope_name}`")
+                self._ws.secrets.create_scope(scope_name)
+            except Exception as ex:
+                logger.error(f"Exception while creating Scope: {ex}")
+                raise ex
+
+            logger.info(f" Created a new Scope: `{scope_name}`")
+        logger.info(f" Using Scope: `{scope_name}`...")
+
+    def secret_key_exists(self, scope_name: str, secret_key: str) -> bool:
```
Comment (nfx) [inline src/databricks/labs/remorph/helpers/db_workspace_utils.py:49]:
```suggestion
    def _secret_key_exists(self, scope_name: str, secret_key: str) -> bool:
```
make private

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    def delete_secret(self, scope_name: str, secret_key: str):
+        try:
+            logger.debug(f"Deleting Secret: *{secret_key}* in Scope: `{scope_name}`")
+            self._ws.secrets.delete_secret(scope=scope_name, key=secret_key)
+        except Exception as ex:
+            logger.error(f"Exception while deleting Secret `{secret_key}`: {ex}")
+            raise ex
+
+    def store_secret(self, scope_name: str, secret_key: str, secret_value: str):
```
Comment (nfx) [inline src/databricks/labs/remorph/helpers/db_workspace_utils.py:66]:
```suggestion
    def _store_secret(self, scope_name: str, secret_key: str, secret_value: str):
```
make private

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+        logger.debug(f"Storing `{engine}` Connection Secrets in Scope: `{scope_name}`")
+
+        for key, value in secrets.items():
+            secret_key = engine + '_' + key
+            logger.debug(f"Processing Secret: *{secret_key}*")
+            if self.secret_key_exists(scope_name, secret_key):
+                overwrite_secret = self._prompts.confirm(f"Do you want to overwrite `{secret_key}`?")
+                if overwrite_secret:
+                    self.delete_secret(scope_name, secret_key)
```
Comment (nfx) [inline src/databricks/labs/remorph/helpers/db_workspace_utils.py:86]: why do you need to delete a secret, if store secret overwrites it?...

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    logger.debug(f"Deleted Secret: *{secret_key}* in Scope: `{scope_name}`")
+                    self.store_secret(scope_name, secret_key, value)
+                    logger.info(f"Overwritten Secret: *{secret_key}* in Scope: `{scope_name}`")
+            else:
+                self.store_secret(scope_name, secret_key, value)
+                logger.info(f"Stored Secret: *{secret_key}* in Scope: `{scope_name}`")
+
+    @property
+    def ws(self):
```
Comment (nfx) [inline src/databricks/labs/remorph/helpers/db_workspace_utils.py:95]: remove, this method is irrelevant.

File: src/databricks/labs/remorph/helpers/recon_config_utils.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    SourceType.DATABRICKS.value,
+    SourceType.NETEZZA.value,
+]
+
+
+class ReconConfigPrompts:
+    def __init__(self, ws: WorkspaceClient, prompts: Prompts = Prompts()):
+        self._source = None
+        self._prompts = prompts
+        self._db_secrets = DatabricksSecretsClient(ws, prompts)
```
Comment (nfx) [inline src/databricks/labs/remorph/helpers/recon_config_utils.py:23]: move all methods from this class over here

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py (hunk 1 above)
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/helpers/db_workspace_utils.py:10]: Done, refactored the code

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py (hunk 2 above)
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/helpers/db_workspace_utils.py:27]: Done,

File: src/databricks/labs/remorph/helpers/db_workspace_utils.py (hunk 6 above)
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/helpers/db_workspace_utils.py:95]: Done

File: src/databricks/labs/remorph/helpers/recon_config_utils.py (hunk 7 above)
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/helpers/recon_config_utils.py:23]: Done

**Review by nfx — APPROVED** [review]
Lgtm

**Pull request merged**